pkg/kubectl/cmd/util/config_flags.go

```diff
@@ -0,0 +1,105 @@
+package util
+
+import (
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+
+	"k8s.io/kubernetes/pkg/kubectl/util/transport"
+)
+
+// ConfigFlags composes the set of values necessary
+// for obtaining a REST client config
+type ConfigFlags struct {
+	CacheDir   *string
+	KubeConfig *string
+}
+
+// ToClientConfig returns a REST client configuration based on a provided
+// path to a .kubeconfig file, loading rules, and config flag overrides.
+// Expects the AddFlags method to have been called.
+func (f *ConfigFlags) ToRESTConfig() (*rest.Config, error) {
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	// use the standard defaults for this client command
+	// DEPRECATED: remove and replace with something more accurate
+	loadingRules.DefaultClientConfig = &clientcmd.DefaultClientConfig
+
+	if f.KubeConfig != nil {
+		loadingRules.ExplicitPath = *f.KubeConfig
+	}
+
+	overrides := &clientcmd.ConfigOverrides{ClusterDefaults: clientcmd.ClusterDefaults}
+	clientConfig := clientcmd.NewInteractiveDeferredLoadingClientConfig(loadingRules, overrides, os.Stdin)
+
+	return clientConfig.ClientConfig()
+}
+
+// ToDiscoveryClient returns a CachedDiscoveryInterface using a computed RESTConfig.
+// Expects the AddFlags method to have been called.
+func (f *ConfigFlags) ToDiscoveryClient() (discovery.CachedDiscoveryInterface, error) {
```

Comment:
@deads2k do we want this method? Do we want to just provide a discovery factory? It appears that we need one of those [to be able to obtain a client cache](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/util/clientcache.go#L32), to then [obtain a kube clientset](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/util/factory_client_access.go#L204).

Unless we want a different approach for obtaining a clientset from this struct?

Reply:
> @deads2k do we want this method? Do we want to just provide a discovery factory? It appears that we need one of those to be able to obtain a client cache, to then obtain a kube clientset.

the cache is gone